server/src/printers/TestMakefilesPrinter.cpp

```diff
@@ -0,0 +1,93 @@
+        fs::path sharedMakefilePath = get_shared(path);
+        FileSystemUtils::writeToFile(sharedMakefilePath, sharedMakefileStr);
+        fs::path objMakefilePath = get_obj(path);
+        FileSystemUtils::writeToFile(objMakefilePath, objMakefileStr);
+    }
+
+    Makefiles::Makefiles(fs::path path, std::string generalMakefileStr, std::string sharedMakefileStr,
+                         std::string objMakefileStr) :
+            path(std::move(path)), generalMakefileStr(std::move(generalMakefileStr)),
+            sharedMakefileStr(std::move(sharedMakefileStr)), objMakefileStr(std::move(objMakefileStr)) {
+    }
+
+    TestMakefilesPrinter::TestMakefilesPrinter(const BaseTestGen &testGen,
+                                               CollectionUtils::FileSet const *stubSources) :
+            projectContext(testGen.projectContext),
+            sharedMakefilePrinter(testGen, stubSources),
+            objMakefilePrinter(testGen, stubSources) {
+    }
+
+    void
+    TestMakefilesPrinter::addLinkTargetRecursively(const fs::path &unitFile, const string &suffixForParentOfStubs) {
+        sharedMakefilePrinter.addLinkTargetRecursively(unitFile, suffixForParentOfStubs, true);
+        objMakefilePrinter.addLinkTargetRecursively(unitFile, suffixForParentOfStubs, false);
+    }
+
+    void TestMakefilesPrinter::close() {
+        sharedMakefilePrinter.close();
+        objMakefilePrinter.close();
+    }
+
+    void TestMakefilesPrinter::init() {
+        sharedMakefilePrinter.init();
+        objMakefilePrinter.init();
+    }
+
+    void TestMakefilesPrinter::addStubs(const CollectionUtils::FileSet &stubsSet) {
+        sharedMakefilePrinter.addStubs(stubsSet);
+        objMakefilePrinter.addStubs(stubsSet);
+    }
+
+    Makefiles TestMakefilesPrinter::GetMakefiles(const fs::path &path) const {
+        printer::DefaultMakefilePrinter generalMakefilePrinter;
+        fs::path generalMakefilePath = Paths::getMakefilePathFromSourceFilePath(projectContext, path);
+
+        generalMakefilePrinter.declareTarget(FORCE, {}, {});
+
+        generalMakefilePrinter.declareTarget("bin", {FORCE}, {
+                StringUtils::joinWith(
+                        MakefileUtils::getMakeCommand(get_shared(generalMakefilePath).string(), "bin", false),
+                        " ")
+        });
+        generalMakefilePrinter.declareTarget("run", {FORCE}, {
+                StringUtils::stringFormat("%s || %s",
+                                          StringUtils::joinWith(MakefileUtils::getMakeCommand(
+                                                  get_shared(generalMakefilePath).string(), "run", false), " "),
+                                          StringUtils::joinWith(
+                                                  MakefileUtils::getMakeCommand(get_obj(generalMakefilePath).string(),
+                                                                                "run", false), " "))
+        });
```

Comment:
What happens if the first command returns exit status `1`? I suppose this is the place to be when the test fails and the framework returns `1`